src/libostree/ostree-repo-pull-verify.c

```diff
@@ -150,31 +150,19 @@ _sign_verify_for_remote (OstreeRepo *repo,
                           GError **error)
 {
   /* list all signature types in detached metadata and check if signed by any? */
-  g_auto (GStrv) names = ostree_sign_list_names();
+  g_autoptr(GPtrArray) signers = ostree_sign_get_all ();
   guint n_invalid_signatures = 0;
   guint n_unknown_signatures = 0;
   g_autoptr (GError) last_sig_error = NULL;
   gboolean found_sig = FALSE;
 
-  for (char **iter=names; iter && *iter; iter++)
+  for (guint i = 0; i < signers->len; i++)
     {
-      g_autoptr (OstreeSign) sign = NULL;
-      g_autoptr (GVariant) signatures = NULL;
-      const gchar *signature_key = NULL;
-      GVariantType *signature_format = NULL;
-
-      if ((sign = ostree_sign_get_by_name (*iter, NULL)) == NULL)
-        {
-          n_unknown_signatures++;
```

Comment:
Also I think previously I said something like we were parsing the list of signatures from the commit object - in doing this commit I realized I was wrong.  

But we have a deeper problem which is that because we've mixed in "signapi signatures" with other metadata, we can't do anything like say "this commit was signed, but with an engine we don't know about".

Oh hmm, I guess we could actually check the metadata to see if there's any `ostree.sign.*` keys that we don't know about.  So yeah this is fine, nevermind!